x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/utils.ts

```diff
@@ -19,3 +19,11 @@ export const getStatusColor = (status: number) => {
   if (status >= 500) color = 'danger';
   return color;
 };
+
+export const safeJsonParseAndStringify = (json: string) => {
+  try {
+    return JSON.stringify(JSON.parse(json), null, 2);
+  } catch {
+    return json;
+  }
+};
```

Comment:
👍 I'm good with adding these comments! The standalone UI had a comment attached to this as well, including it here for historical context

```js
// parses, then stringifies json content
// If the content isn't properly formatted JSON, the original content is returned
```

On a philosophical / smol soapbox aside:

> so one doesn't have to open the test file to work out what this is accomplishing

My 2c on this is are along the lines of "it's a feature, not a bug". My original thinking for skipping the comment was because the unit tests did document it so clearly. Documenting code design and intention is definitely one of the foremost purposes of unit testing (I've occasionally referred to it as "living documentation"), and devs should definitely default to opening unit tests to understand code when uncommented, especially when checking branches.

That being said, I'm definitely not overly didactic about the concept, nor am I in the militant "comments are a code smell, code should be self-documenting" camp and I definitely agreed this fn isn't as straightforward as the others, so happy to add the suggested comments!